examples/hexy.rs

```diff
@@ -9,7 +9,7 @@ use std::fmt;
 use std::str::FromStr;
 
 use hex_conservative::{
-    fmt_hex_exact, Case, DecodeFixedSizedBytesError, DisplayHex as _, FromHex as _,
+    fmt_hex_exact, Case, DecodeFixedLengthError, DisplayHex as _, FromHex as _,
```

Comment:
It sounds like decoding som kind of length but it's not decoding length, it's decoding bytes. I'd suggest `DecodeFixedLengthBytes` or `DecodeFixedLenBytes`.

Regarding commit message, my main argument is that if the caller then returns the error it "leaks" the type being decoded. E.g. We could later have a function to decode into `Box<[u8]>` so `DecodeToVec` would be misleading.